models/articles.model.js

```diff
@@ -9,25 +8,65 @@ exports.selectArticlesById = (article_id) => {
     });
 };
 
-exports.selectAllArticles = () => {
-    let query = `SELECT 
-    articles.author, 
-    title, 
-    articles.article_id, 
-    topic, 
-    articles.created_at, 
-    articles.votes, 
-    articles.article_img_url, 
-    COUNT(comments.author) AS comments_count  
-    FROM articles 
-    LEFT JOIN comments ON comments.article_id = articles.article_id 
-    GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC
-    ;
-    `;
-    return db.query(query).then(({ rows }) => {
-        return rows;
-    });
+exports.selectAllArticles = async (topic, sort_by, order) => {
+    const fetchingTopicsInArticles = await db
+        .query(`SELECT DISTINCT topic FROM articles;`)
+        .then(({ rows }) => {
+            let topics = rows.map((row) => row.topic);
+            return topics;
+        });
+
+    let greenList = [
+        "desc",
+        "asc",
+        ...fetchingTopicsInArticles,
+        "author",
+        "title",
+        "article_id",
+        "topic",
+        "created_at",
+        "votes",
+        "article_img_url",
+    ];
+
+    let query = `SELECT articles.author, title, 
+                 articles.article_id, topic, 
+                 articles.created_at, articles.votes, articles.article_img_url, 
+                 COUNT(comments.author) AS comments_count  
+                 FROM articles 
+                 LEFT JOIN comments ON comments.article_id = articles.article_id `;
+
+    const queryValues = [];
+
+    if (topic) {
+        if (!greenList.includes(topic)) {
+            return db.query(`select * from topics where slug = $1`, [topic]).then(({ rows }) => {
+                return !rows.length
+                    ? Promise.reject({ status: 400, msg: "Bad request" })
+                    : Promise.reject({ status: 404, msg: "Not found!" });
+            });
+        }
```

Comment:
There is a missing happy path test about a valid topic that doesn't have any articles associated with it - this will change the code in this model a bit.

In lines 42-48 you're over-engineering a bit. Why not just make all topics greenlisted in the first place rather than just selecting topics from articles? This question might be more relevant now you know we want a different response if the topic is valid but doesn't have any articles